corehq/apps/linked_domain/keywords.py

```diff
@@ -4,9 +4,17 @@
 
 from corehq.apps.linked_domain.applications import get_downstream_app_id
 from corehq.apps.linked_domain.exceptions import DomainLinkError, MultipleDownstreamAppsError
+from corehq.apps.sms.dbaccessors import get_keywords_for_domain
 from corehq.apps.sms.models import Keyword
 
 
+def get_downstream_keyword(downstream_domain, upstream_keyword_id):
+    for keyword in get_keywords_for_domain(downstream_domain):
+        if keyword.upstream_id == str(upstream_keyword_id):
```

Comment:
Stylistic suggestion: I'd have given `get_keywords_for_domain` an optional `upstream_id` arg, just so it could be passed directly as a filter and avoid both the loop and the `str` cast.